include/envoy/network/connection.h

```diff
@@ -120,6 +120,7 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
    * @return The address of the remote client.
    */
   virtual const Address::Instance& remoteAddress() PURE;
+  virtual Address::InstanceConstSharedPtr remoteAddressConstSharedPtr() PURE;
```

Comment:
Per hidden comment, very confused why the new functions have been added. Just putting this here so it shows up in PR.